tests/components/device_tracker/test_asuswrt.py

```diff
@@ -0,0 +1,51 @@
+"""The tests for the ASUSWRT device tracker platform."""
+from homeassistant.components.asuswrt import (
+    CONF_PROTOCOL, CONF_MODE, DOMAIN, CONF_PORT, DATA_ASUSWRT, async_setup)
+from homeassistant.const import (CONF_PLATFORM, CONF_PASSWORD, CONF_USERNAME,
+                                 CONF_HOST)
+
+from tests.common import MockDependency, mock_coro_func
+
+FAKEFILE = None
+
+VALID_CONFIG_ROUTER_SSH = {DOMAIN: {
+    CONF_PLATFORM: 'asuswrt',
+    CONF_HOST: 'fake_host',
+    CONF_USERNAME: 'fake_user',
+    CONF_PROTOCOL: 'ssh',
+    CONF_MODE: 'router',
+    CONF_PORT: '22'
+}}
+
+
+async def test_password_or_pub_key_required(hass):
+    """Test creating an AsusWRT scanner without a pass or pubkey."""
+    with MockDependency('aioasuswrt.asuswrt')as mocked_asus:
+        mocked_asus.AsusWrt().connection.async_connect = mock_coro_func()
+        mocked_asus.AsusWrt().is_connected = False
+        assert not await async_setup(
```

Comment:
Use `async_setup_component` from homeassistant.setup.py. Look at other tests for examples.